pkg/reconciler/instances/istio/actions/performer.go

```diff
@@ -75,52 +77,66 @@ type DataPlaneVersion struct {
 	IstioVersion string `json:"IstioVersion,omitempty"`
 }
 
+type CommanderResolver interface {
+	GetCommander(version istioctl.Version) (istioctl.Commander, error)
+}
+
 //go:generate mockery --name=IstioPerformer --outpkg=mock --case=underscore
 // IstioPerformer performs actions on Istio component on the cluster.
 type IstioPerformer interface {
 
-	// Install Istio on the cluster.
-	Install(kubeConfig, manifest string, logger *zap.SugaredLogger) error
+	// Install Istio on the cluster. Tries to find best matching istioctl binary for the 'requestedVersion'.
+	Install(kubeConfig, istioChart, requestedVersion string, logger *zap.SugaredLogger) error
```

Comment:
Should we use the term `requestedVersion`? In `Istioversion` struct we keep it as a `targetVersion`. I find it as an inconsistency.

Reply:
I changed that because initially I didn't understood what a "targetVersion" is - I've had to carefully read the code. I think in this high-level interface we should use names that are easy to understand.
To support that,  I've changed the names again - please take a look. They are not the same in every method though, but I think they are understandable even for people outside our domain.

Reply:
Discussed.